policy/modules/kernel/domain.if

```diff
@@ -1487,6 +1487,32 @@ interface(`domain_all_recvfrom_all_domains',`
 	corenet_all_recvfrom_labeled($1, domain)
 ')
 
+########################################
+## <summary>
+##	Allow all domains to search specifyed type keys.
+## </summary>
+## <desc>
+##	<p>
+##	When setting up IMA/EVM key(s) are added to the 
+##	kernel keyring but the type of the key is the domain
+##	adding the key.  This interface will allow all domains
+##	search the key so IMA/EVM validation can happen.
+##	</p>
+## </desc>
+## <param name="type">
+##	<summary>
+##	Type of key to be searched.
+##	</summary>
+## </param>
+#
+interface(`domain_all_search_key',`
```

Comment:
`domain_search_all_keys()`

Reply:
Note this is a reverse interface: the argument is used as the target context.

I'd suggest something like `domain_public_key`

Reply:
I agree, it is allowing all domains to search keys with a specific type.  I don't think that 'search_all_keys' is correct.  Go with 'domain_public_key' or something else?

Reply:
@cgzones thanks for pointing that out; I missed that detail on my first review. `domain_public_key()` is more appropriate, though I'm still not a fan of it.

Reply:
I have made the change to the name.  I agree, I'm not a big fan of the name, feel free to suggest something else before merging if you can think of something better. The original name was trying to suggest that all domains could search keys of the type provided. 

Another choice (that I think I'm less a fan of) is to update domain.te around line 96 to change the 'kernel_dontaudit_*_key(domain)' to make these allow rules.  Then I put the keys into the kernel keyring with kernel_t.  It will require an interface 'kernel_create_key' (I think) and maybe something else (I can't remember). 

I think the submitted approach is more clear where the keys came from.